python-checks/src/main/java/org/sonar/python/checks/UnusedLocalVariableCheck.java

```diff
@@ -138,10 +141,30 @@ public PreciseIssue createIssue(SubscriptionContext ctx, Symbol symbol, Usage us
     } else {
       var issue = ctx.addIssue(usage.tree(), String.format(MESSAGE, symbol.name()));
       createExceptClauseQuickFix(usage, issue);
+      createAssignmentQuickFix(usage, issue);
       return issue;
     }
   }
 
+  private static void createAssignmentQuickFix(Usage usage, PreciseIssue issue) {
+    if (usage.kind().equals(Usage.Kind.ASSIGNMENT_LHS)) {
+      Statement assignmentStatement = ((Statement) TreeUtils.firstAncestorOfKind(usage.tree(), Kind.ASSIGNMENT_STMT, Kind.ANNOTATED_ASSIGNMENT));
+
+      Optional.ofNullable(assignmentStatement).filter(stmt -> stmt.is(Kind.ASSIGNMENT_STMT)).map(AssignmentStatement.class::cast).ifPresent(stmt -> {
+        PythonQuickFix quickFix = PythonQuickFix.newQuickFix(ASSIGNMENT_QUICK_FIX_MESSAGE,
+          TextEditUtils.removeUntil(usage.tree(), stmt.assignedValue().firstToken()));
+        issue.addQuickFix(quickFix);
+      });
+
+      Optional.ofNullable(assignmentStatement).filter(stmt -> stmt.is(Kind.ANNOTATED_ASSIGNMENT)).map(AnnotatedAssignment.class::cast)
+        .map(AnnotatedAssignment::assignedValue).ifPresent(stmt -> {
+          PythonQuickFix quickFix = PythonQuickFix.newQuickFix(ASSIGNMENT_QUICK_FIX_MESSAGE,
+            TextEditUtils.removeUntil(usage.tree(), stmt.firstToken()));
```

Comment:
_Nitpick:_ I guess here `stmt` is no longer the statement but the assigned value (if present), I think it'd be better if the variable name reflected that.